catalog/resource_grants.go

```diff
@@ -374,7 +389,12 @@ func ResourceGrants() common.Resource {
 			var grants PermissionsList
 			common.DataToStructPointer(d, s, &grants)
 			unityCatalogPermissionsAPI := permissions.NewUnityCatalogPermissionsAPI(ctx, c)
-			return replaceAllPermissions(unityCatalogPermissionsAPI, securable, name, grants.toSdkPermissionsList())
+			err = replaceAllPermissions(unityCatalogPermissionsAPI, securable, name, grants.toSdkPermissionsList())
+			if err != nil {
+				return err
+			}
+			d.SetId(mapping.id(d))
```

Comment:
This was another bug, we don't update id when changing name of underlying resource

Reply:
Is this actually a bug? When do we need to change the ID? If either the securable type or securable itself changes, the resource will be recreated, so this pathway is not taken.